src/components/ChatEntry.js

```diff
@@ -1,22 +1,30 @@
-import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
+
 
 const ChatEntry = (props) => {
+  const buttonContent = props.liked ? '❤️' : '🤍';
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={props.sender === 'Estragon' ? 'chat-entry remote' : 'chat-entry local'}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp} /></p>
+        <button className="like" onClick={() => props.onSetLike(props.id)}>{buttonContent}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  id: PropTypes.number.isRequired,
+  onSetLike: PropTypes.func
```

Comment:
Leaving this not marked `isRequired` avoids a bunch of warnings when running the tests (the tests don't know what you're going to call the event handlers). Ideally, we'd also include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (props.onSetLike) {
        props.onSetLike(props.id);
    }
```
which would only try to call the function if it's not falsy (and `undefined` is falsy).

(We might want to move the click logic out of an inline function if we added this).